### ISSUE TITLE
Model Compiler - NNIR FP16 Conversion

### ISSUE THREAD

**Issue by kiritigowda** [GPUOpen-ProfessionalCompute-Libraries/MIVisionX#429]
FP16 Conversion broken

```
ERROR: invalid data (magic,size)=(0x%x,%d) in %s at byte position %d -- expected size is %ld
ERROR: failed with status = (-1) at annmodule.cpp#119
ERROR: annAddToGraph() failed (-1)
```

**Comment by kiritigowda** [comment]
@LakshmiKumar23 changes to model compiler have to pass these tests - https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/tree/master/tests/neural_network_tests#mivisionx-neural-network-tests

Look through and see if there are any gaps.

**Comment by kiritigowda** [comment]
@LakshmiKumar23 any updates on this?

**Comment by kiritigowda** [comment]
@paveltc can you point to the commit which starting to fail the FP16 flow?

**Comment by LakshmiKumar23** [comment]
@kiritigowda @paveltc #436 fixes this

**Comment by kiritigowda** [comment]
Fixed with pr #436
